database/migrations/2021_03_08_031840_create_posts_table.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+class CreatePostsTable extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('posts', function (Blueprint $table) {
+            $table->increments('id');
+            $table->integer('user_id')->unsigned()->index();
+            $table->foreign('user_id')->references('id')->on('users');
```

Comment:
These two line seem to be creating the same column; one or the other would do.

Reply:
To be honest I don't even use foreign key constraints, although the second line probably does what you want. 

You can and should validate the existence of any database IDs within the app where it is coming from user input.

In this instance `request()->user()->id` is validated by Laravel earlier in the request lifecycle so we shouldn't consider it to be user input.

Reply:
My apologies I don't understand much of the constraints and laravel validation, I'll make sure to read more about that as well!